handlers.go

```diff
@@ -81,3 +81,40 @@ func checksHandler(c *echo.Context) error {
 
 	return c.JSON(http.StatusOK, checks)
 }
+
+// DELETE /sites/:key
+func deleteSiteHandler(c *echo.Context) error {
+	siteKey := c.P(0)
+
+	_, err := db.Exec("DELETE FROM sites WHERE id=$1", siteKey)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Error deleting site")
+	}
+
+	_, err = db.Exec("DELETE FROM checks WHERE siteId=$1", siteKey)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Error deleting site checks")
+	}
+
+	removeSite(siteKey)
+
+	return c.NoContent(http.StatusNoContent)
+}
+
+// Helper functions
+func removeSiteByIndex(i int) {
+	if i < len(AllSites) - 1 {
+		AllSites = append(AllSites[:i], AllSites[i + 1:]...)
+    } else {
+        AllSites = AllSites[:i]
+    }
+}
+
+func removeSite(Id string) {
+	for i := range AllSites {
```

Comment:
Should we change `AllSites` to be a hash map for easier lookup and deletion instead of just a plain old array?